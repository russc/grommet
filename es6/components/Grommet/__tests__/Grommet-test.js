function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { cleanup, render } from 'react-testing-library';
import { hpe as hpeTheme } from 'grommet-theme-hpe';
import { Grommet } from '..';
import { AnnounceContext } from '../../../contexts';

var TestAnnouncer =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(TestAnnouncer, _Component);

  function TestAnnouncer() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = TestAnnouncer.prototype;

  _proto.componentDidMount = function componentDidMount() {
    var announce = this.props.announce;
    announce('hello', 'assertive');
  };

  _proto.render = function render() {
    return React.createElement("div", null, "hi");
  };

  return TestAnnouncer;
}(Component);

_defineProperty(TestAnnouncer, "propTypes", {
  announce: PropTypes.func.isRequired
});

describe('Grommet', function () {
  afterEach(cleanup);
  test('basic', function () {
    var component = renderer.create(React.createElement(Grommet, null));
    expect(component.toJSON()).toMatchSnapshot();
  });
  test('hpe theme', function () {
    var component = renderer.create(React.createElement(Grommet, {
      theme: hpeTheme
    }, "Grommet App"));
    expect(component.toJSON()).toMatchSnapshot();
  });
  test('full', function () {
    var component = renderer.create(React.createElement(Grommet, {
      full: true
    }, "Grommet App"));
    expect(component.toJSON()).toMatchSnapshot();
  });
  test('announce', function (done) {
    var _render = render(React.createElement(Grommet, null, React.createElement(AnnounceContext.Consumer, null, function (announce) {
      return React.createElement(TestAnnouncer, {
        announce: announce
      });
    }))),
        container = _render.container;

    expect(container.firstChild).toMatchSnapshot(); // no style, no need for expectPortal

    expect(document.body.querySelector('[aria-live]')).toMatchSnapshot();
    setTimeout(function () {
      // should clear the aria-live container
      expect(document.body.querySelector('[aria-live]')).toMatchSnapshot();
      done();
    }, 600); // wait the aria-live container to clear
  });
});