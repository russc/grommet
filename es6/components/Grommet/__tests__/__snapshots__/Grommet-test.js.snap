// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grommet announce 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <div>
    hi
  </div>
</div>
`;

exports[`Grommet announce 2`] = `
<div
  aria-live="assertive"
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
>
  hello
</div>
`;

exports[`Grommet announce 3`] = `
<div
  aria-live="assertive"
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
/>
`;

exports[`Grommet basic 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
/>
`;

exports[`Grommet full 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  className="c0"
>
  Grommet App
</div>
`;

exports[`Grommet hpe theme 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  Grommet App
</div>
`;
